### ISSUE TITLE
getPedAnimation() returns false in some situations

### ISSUE THREAD

**Issue by Einheit-101** [multitheftauto/mtasa-blue#954]
**Describe the bug**
I am doing many experiments with animations currently, and i found yet another bug.
getPedAnimation() can be bugged with a simple code, when calling setPedAnimation() in multiple frames behind each other.

**The following code will make the player move backwards forever.
He should actually stop moving because after 30ms (2 frames later) we call "idle_1armed" animation, which should overwrite "gunmove_bwd". Additionally, getPedAnimation() will return false as if no animation is playing at all.**

```
function bug()
setPedAnimation(localPlayer, "ped", "gunmove_r", -1, true, true, false, false, 50, true)
setTimer(function()
	setPedAnimation(localPlayer, "ped", "gunmove_bwd", -1, true, true, false, false, 50, true)
end, 15, 1)
setTimer(function()
	setPedAnimation(localPlayer, "ped", "idle_1armed", 50, false, true, true, false, 50, false)
end, 30, 1)
end
addCommandHandler("bug", bug)
```

**Expected behaviour**
The player should correctly stop moving and play "idle_1armed" animation after 30 milliseconds.
getPedAnimation() should not return false after using the above code.

**Multi Theft Auto v1.5.6-release-18427**

**Comment by Einheit-101** [comment]
@qaisjp i think this should be fixed either

**Comment by ghost** [comment]
@Einheit-101 I can't reproduce it in latest nightly. I think this is related to #947. Some code from restoreTaskOnAnimEnd was causing a lot of issues, so I removed it in 65ae16887ae374b1e9b748436f244d2502ad2869. 

Can you check if the bug is still there?

**Comment by qaisjp** [comment]
I have modified the above code to look like this:

```lua
function bug()
	local result = setPedAnimation(localPlayer, "ped", "gunmove_r", -1, true, true, false, false, 50, true)
	iprint("one", result)
	setTimer(function()
		local result = setPedAnimation(localPlayer, "ped", "gunmove_bwd", -1, true, true, false, false, 50, true)
		iprint("two", result)
	end, 15, 1)
	setTimer(function()
		local result = setPedAnimation(localPlayer, "ped", "idle_1armed", 50, false, true, true, false, 50, false)
		iprint("three", result)
	end, 30, 1)
end
addCommandHandler("bug", bug)
```

- On 10cd2edcb (this is before 65ae168)
    - when standing still, I keep moving back, and output is `true, true, true`
    - whilst the previous animation is running (from a previous run of `/bug`), after running the command, I stop moving back, and output is `true, true, true`
- On a389d5290991f2e39317b59c0e7dbc131c10228c (this is after 65ae168)
    - behaves identically to 65ae168
    - but I managed to reproduce the animation stopping **once**

tl;dr I can't reproduce it returning false, and in most scenarios the player continuously moves backwards. FYI this is client-side. @Einheit-101 are you running this server-side?

Might also be worth testing this behaviour if we revert the retain ped state feature

**Comment by Einheit-101** [comment]
I tried to reproduce it again and i can still reproduce it, but it doesnt work always.
It seems to behave correctly for **one** time, as qaisjp said, but it fails everytime after that.

**Comment by Einheit-101** [comment]
@saml1er This bug still works in at least 1 of 3 cases, i simply add the reproduction code and type "bug" a few times and the player keeps moving backwards at some point.

**Comment by FileEX** [comment]
There is no animation called `idle_1armed` - it should be `idle_armed`.

**Comment by Einheit-101** [comment]
> There is no animation called `idle_1armed` - it should be `idle_armed`.

That animation actually exists, it may not be in the default folder bc it may be used by ai, i dont know if this bug still exists tho

**Comment by FileEX** [comment]
> > There is no animation called `idle_1armed` - it should be `idle_armed`.
> 
> That animation actually exists, it may not be in the default folder bc it may be used by ai, i dont know if this bug still exists tho

This animation is not listed anywhere in any animation list. It may exist in later versions of GTA than 1.0
